include/revng/Support/ManagedStaticRegistry.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <compare>
+#include <set>
+
+#include "llvm/ADT/Twine.h"
+#include "llvm/Support/ManagedStatic.h"
+
+#include "revng/Support/Debug.h"
+
+template<typename Type>
+concept IsStaticallyRegisterable = requires(Type &&Value) {
+  { Value.key() } -> std::convertible_to<llvm::StringRef>;
+};
+
+template<IsStaticallyRegisterable Registered>
+class RegisterManagedStaticImpl {
```

Comment:
Can we drop this in favor of `pipeline::Registry`?
There seem to be an overlap.

Reply:
I don't want to protract this PR much longer, but take a look if it's easily doable.

Reply:
Pipeline uses a custom solution that does not rely on `llvm::ManagedStatic`.

Considering that it stores a bunch of different entities in different containers, conversion is not going to be trivial. We could, maybe, try to do something with the root of it all at least (`include/revng/Support/DynamicHierarchy.h`), but IMO, delaying this PR any more is not worth it.

Let the `DynamicHierarchy.h` go away when `pypeline` is ready.

Maybe it's more worth looking into `Tag`s, the only non-pipeline user of `DynamicHierarchy.h`, but again, let future us worried about that, this PR already has way to much non-name-related logic stappled to it :D